vpn/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldTrackerActivity.kt

```diff
@@ -156,16 +157,7 @@ class DeviceShieldTrackerActivity :
         data: Intent?
     ) {
         if (requestCode == REQUEST_ASK_VPN_PERMISSION) {
-            when (resultCode) {
-                RESULT_OK -> {
-                    startVPN()
-                    return
-                }
-                else -> {
-                    deviceShieldSwitch.quietlySetIsChecked(false, enableAppTPSwitchListener)
-                    Timber.d("Permission not granted")
-                }
-            }
+            viewModel.onVPNPermissionResult(resultCode)
```

Comment:
This is not working as before (see view model)
* in the case of `RESULT_OK`. Before we were `startingVPN()` and returning, now we are calling `super.onActivityResult(requestCode, resultCode, data)`
* In the `else` case we are now calling `stopDeviceShield()` in case the permission is not granted...why?